### ISSUE TITLE
LDAP2 causes Segmentation Fault in 8.6 ([ERROR] Received signal 11, going down)

### ISSUE THREAD

**Issue by talseinfeld** [vanhauser-thc/thc-hydra#380]
Hey,
I've tried to use Hydra with ldap in Hydra 8.6 and I always got a segmentation fault.
I added two flags to the configuraton: --nostrip and --debug; afterwards I've changed the -O3 flag in the Makefile to -g. Then I got the following stacktrace:
![gdb_stack_hydra](https://user-images.githubusercontent.com/25203317/49699906-e7cc3a00-fbdf-11e8-9940-25eeb9f542b7.jpg)
I marked the miscptr which is null. I then went to edit the method that caused the segmentation fault to allocate miscptr if it's null, just to bypass the seg fault and see if I get an ldap authentication correctly - Unfortunately, I didn't:
![image](https://user-images.githubusercontent.com/25203317/49699930-3f6aa580-fbe0-11e8-8e1d-43c6d4c6e400.png)
Here is the Wireshark information showing ldap received bad credentials (though I've entered correct credentials):
![wireshark_ldap_errpr](https://user-images.githubusercontent.com/25203317/49699941-58735680-fbe0-11e8-9763-f00a5553dd87.jpg)

I might have ruined further parsing or handling to the miscptr by allocating it whether it is null?
P.S
SMB authentication to the same server works just fine with Hydra

Thanks in advance

**Comment by vanhauser-thc** [comment]
are you on linux or mac?

have you tried to pass -m "" ?

but its a bug, will have to look into it

**Comment by vanhauser-thc** [comment]
I merged a fix, can you please try again?

for windows ad auth: you must use ldaps (ssl) for it, thats for sure, and likely ldap2 or ldap3, and add @domain.org to the username. plus it would depend on the cn= etc. definitions ... see maybe here: https://docs.geoserver.org/latest/en/user/security/tutorials/activedirectory/index.html

and then (after merging and compiling) look at this: 
```
  hydra -U ldap2
```

**Comment by talseinfeld** [comment]
OK - First of all it still gives segmentation fault error when given just login argument; e.g.: 'hydra -l username -p password -vV ldap2://dcip:389'. It only works if I give it explicit CN=Full user name,CN=Users,DC=domain, DC=org. The only thing I know at the moment I'm trying to use Hydra is the logon username (the one that works via SMB) and his password. Is there any way to use the user's logon name via the login argument, without using the AD parameters in the miscptr string? Or is there way to forward an AD parameter that is the user logon name (couldn't find anything that works except the full user name; e.g.: First name initials. surname).
I just want to validate that the credentials I've found during a penetration test are valid via LDAP (if I know I am targeting a DC). Regarding the seg fault - I think that if you'd enable support for using hydra with -l and with -m it would solve the null pointer, or you will find where it throws it again.

**Comment by vanhauser-thc** [comment]
I dont have an ldap server so I can not test this. until the point where the ldap server would reply there is no segfault for me.

can you please:
1. ensure that you checked out the newest version
2. make clean
3. ./configure
4. vi Makefile  (and add -g to OPTS=...)
5. make
6. gdb ./hydra
 and then "run -l login -p pw" ... etc that makes it segfault
and then show the backtrace please.

**Comment by talseinfeld** [comment]
Hey Van - Is there any way to get newest hydra via apt or the only way
right now is to git clone it, configure, make and make install?

**Comment by vanhauser-thc** [comment]
you just do "git pull" to update from the repository, then ./configure and make

**Comment by talseinfeld** [comment]
Yeah this is exactly what I've done but suddenly the program says that I
haven't compiled the file with OpenSSL thus I cannot run SMB module. Any
bypass?

**Comment by talseinfeld** [comment]
Just updating that I have managed to fix this issue, but be advised that
for some reason when you git clone, configuring, making and make installing
it might say you have compiled it without openssl. Maybe you should add
openssl to your requirements. I'll check back without the openssl wget full
installation and just with the apt-get install libs you've listed and I'll
keep you posted

בתאריך יום ה׳, 13 בדצמ׳ 2018, 11:31, מאת Tal Sheinfeld <talz.gt@gmail.com>:

**Comment by talseinfeld** [comment]
Installing OpenSSL with wget along with its long installation process
solved it for me; but I'm working on a docker container and not a VM Ubuntu
or such.
Have a great weekend and thank you for your support,

Cheer,s
